docker/entrypoint.sh

```diff
@@ -11,5 +11,5 @@ exec java \
   -Xmx$JAVA_MAX_MEMORY \
   -Djava.net.preferIPv4Stack=true \
   -jar $DOCKER_IRI_JAR_PATH \
-  --remote true --remote-limit-api "$DOCKER_IRI_REMOTE_LIMIT_API" \
+  --remote $DOCKER_IRI_REMOTE --remote-limit-api "$DOCKER_IRI_REMOTE_LIMIT_API" \
```

Comment:
Missing the quotes on `$DOCKER_IRI_REMOTE` -> `"$DOCKER_IRI_REMOTE"`. And we're good to go.

Reply:
@sadjy thanks.
I was thinking to quote the $DOCKER_IRI_REMTOE tbh given this is best practice for bash.
Because this value is a single word (false|true) I have omitted the double quotes.
Will quote it,

How about the variable in `-jar $DOCKER_IRI_JAR_PATH` ? It also missed the double quotes?
For `"$DOCKER_IRI_REMOTE_LIMIT_API"` obviously this was quoted as it might hold a string with spaces in it ...

Reply:
The boolean value is interpreted and "disappears" when the `entrypoint.sh` runs. Leaving the line as `--remote --remote-limit-api "$DOCKER_IRI_REMOTE_LIMIT_API"` and thus failing to run (as `--remote` see `--remote-limit-api` as its argument and cannot make sense of it.)
The `$DOCKER_IRI_JAR_PATH` is a string so it doesn't really matter.
Right on for the the last point!
Thanks :)

Reply:
Hi @sadjy . I have quoted the variable.

Sorry, this is a lengthy message :) I hope you don't mind...
I wanted to reply to your last comment and highlight some of my thoughts.

# Boolean value for --remote

> The boolean value is interpreted and "disappears" when the entrypoint.sh runs. Leaving the line as --remote --remote-limit-api 

I don't believe this is the case. It would have been the case if I would have provided the variable as: `$($DOCKER_IRI_REMOTE)` causing it to actually execute (or wrap in backticks)
It is treated as a string.

I have done a test to provide an example:
Here is the entrypoint.sh I have used where I didn't double quote the variable:
```sh
#!/bin/bash
# See Dockerfile and DOCKER.md for further info

if [ "${DOCKER_IRI_MONITORING_API_PORT_ENABLE}" == "1" ]; then
  nohup socat -lm TCP-LISTEN:14266,fork TCP:127.0.0.1:${DOCKER_IRI_MONITORING_API_PORT_DESTINATION} &
fi

echo HAVE $DOCKER_IRI_REMOTE

exec java \
  $JAVA_OPTIONS \
  -Xms$JAVA_MIN_MEMORY \
  -Xmx$JAVA_MAX_MEMORY \
  -Djava.net.preferIPv4Stack=true \
  -jar $DOCKER_IRI_JAR_PATH \
  --remote $DOCKER_IRI_REMOTE --remote-limit-api "$DOCKER_IRI_REMOTE_LIMIT_API" \
  "$@"
```
And I have the `ENV DOCKER_IRI_REMOTE true` in the Dockerfile.

Running this docker image, we see that IRI has correctly parsed the CLI options:
```sh
[root@x-vps iri]# docker run --rm iotaledger/iri:testing
HAVE true
Logging - property 'logging-level' set to: [INFO]
06/17 18:47:26.975 [main] INFO  IRI$IRILauncher:188 - Configuration is created using command line args only
06/17 18:47:26.982 [main] INFO  IRI$IRILauncher:189 - parsed the following cmd args: [--remote, true, --remote-limit-api, interruptAttachToTangle, attachToTangle, addNeighbors, removeNeighbors, getNeighbors]
06/17 18:47:26.984 [main] INFO  IRI$IRILauncher:119 - Welcome to IRI 1.7.0-DEV-${git.commit.id.abbrev}
```
Notice the `--remote, true,` in the log output. Overriding the ENV with false works as well.
But anyway, I agree it is best practice to double quote bash variables.


# Quoting iri jar path

About `$DOCKER_IRI_JAR_PATH`: It is set by default to `ENV DOCKER_IRI_JAR_PATH "/iri/target/iri*.jar"` which is fine: no spaces.

Nevertheless, for best practices in bash and given we have decided to quote $DOCKER_IRI_REMOTE, it seems to make sense to quote DOCKER_IRI_JAR_PATH too. 

Why is double quoting a path a good practice: what happens if I am a noob who decided to create a path for IRI in the container that has spaces in the name?

In the following example I am going to override the default and set a new path by passing an environment variable to docker:
```-e DOCKER_IRI_JAR_PATH="/some/path/test hello/here"```

I will also override the default entry point script, it isn't much different, and the DOCKER_IRI_JAR_PATH is not quotes, as is currently the case in the official entrypoint script:
```sh
#!/bin/bash
# See Dockerfile and DOCKER.md for further info

if [ "${DOCKER_IRI_MONITORING_API_PORT_ENABLE}" == "1" ]; then
  nohup socat -lm TCP-LISTEN:14266,fork TCP:127.0.0.1:${DOCKER_IRI_MONITORING_API_PORT_DESTINATION} &
fi

echo HAVE $DOCKER_IRI_REMOTE
echo IRI_JAR_PATH: $DOCKER_IRI_JAR_PATH

exec java \
  $JAVA_OPTIONS \
  -Xms$JAVA_MIN_MEMORY \
  -Xmx$JAVA_MAX_MEMORY \
  -Djava.net.preferIPv4Stack=true \
  -jar $DOCKER_IRI_JAR_PATH \
  --remote $DOCKER_IRI_REMOTE --remote-limit-api "$DOCKER_IRI_REMOTE_LIMIT_API" \
  "$@"
```

I run:
```sh
[root@x-vps iri]# docker run --rm -v $(pwd)/docker/entrypoint.sh:/entrypoint.sh -e DOCKER_IRI_JAR_PATH="/some/path/test hello/here" iotaledger/iri:testing
HAVE true
IRI_JAR_PATH: /some/path/test hello/here
Error: Unable to access jarfile /some/path/test
```
As you can see, echoing without double quotes is fine. 
However, the `-jar` only got passed the portion until the space: `/some/path/test`. The message in the error indicates how this was parsed `Error: Unable to access jarfile /some/path/test`.

And then, if I add double quote on the $DOCKER_IRI_JAR_PATH I get:
`Error: Unable to access jarfile /some/path/test hello/here`

Therefore, I think the entry point, (also in sake of consistency) should use double quotes on all the variables.

# Dockerfile improvement

Regarding the Dockerfile, I think that in stead of having a step per ENV, they can all be configured under a single ENV. Notice what happens during the build, we keep adding layers:
```
Step 10/19 : ENV JAVA_OPTIONS="-XX:+UnlockExperimentalVMOptions -XX:+DisableAttachMechanism -XX:InitiatingHeapOccupancyPercent=60 -XX:G1MaxNewSizePercent=75 -XX:MaxGCPauseMillis=10000 -XX:+UseG1GC"
---> Running in e0723608b931
Removing intermediate container e0723608b931
---> 447a631f79f7
Step 11/19 : ENV JAVA_MIN_MEMORY 2G
---> Running in 7f670224ff26
Removing intermediate container 7f670224ff26
---> 8fa2efe529b3
Step 12/19 : ENV JAVA_MAX_MEMORY 4G
---> Running in 180f9807b6dd
Removing intermediate container 180f9807b6dd
---> 4d445d9ab66d
Step 13/19 : ENV DOCKER_IRI_JAR_PATH "/iri/target/iri*.jar"
---> Running in 167bbff9749e
Removing intermediate container 167bbff9749e
---> fc88b51f828f
Step 14/19 : ENV DOCKER_IRI_REMOTE_LIMIT_API "interruptAttachToTangle, attachToTangle, addNeighbors, removeNeighbors, getNeighbors"
---> Running in 974bfa6318c0
Removing intermediate container 974bfa6318c0
---> 3f068d556097
Step 15/19 : ENV DOCKER_IRI_MONITORING_API_PORT_ENABLE 0
---> Running in 4c7daac99151
Removing intermediate container 4c7daac99151
---> b6ffad1d5085
Step 16/19 : ENV DOCKER_IRI_MONITORING_API_PORT_DESTINATION 14265
---> Running in 9f6b8ab95734
Removing intermediate container 9f6b8ab95734
---> a169fe07ed2d
Step 17/19 : ENV DOCKER_IRI_REMOTE true
---> Running in 5dbdd1dfafb7
Removing intermediate container 5dbdd1dfafb7
```

Where as, with a single ENV as in this example:
```
ENV VALUE=true \
    ANOTHER_VALUE=false \
    HELLO=world \
    HI="hi there"
```
The build will have one layer only:
```
Step 3/5 : ENV VALUE=true     ANOTHER_VALUE=false     HELLO=world     HI="hi there"
 ---> Running in 1f2fa5e9a587
Removing intermediate container 1f2fa5e9a587
```
Layers of ENV are 0 bytes size, but I think it would be nicer to put all under one ENV command.
Regarding the IRI Dockerfile comments above some of the ENV lines, they would have to move above.

# Conclusion
If you agree, I can prepare a PR with all the points mentioned here.
Thank you!

Reply:
Hi Nuriel. Great points made all around. I'd appreciate if you take the time to issue such a PR (or you can even use this same PR to issue the mentioned changes if you like). 
Thank you for your response!  We very much appreciate this feedback. :)